### ISSUE TITLE
Logger/ Error / Loggin Support

### ISSUE THREAD

**Issue by ghost** [scg-unibe-ch/ese2016-team8#16]
Following **4. Design** from [Architectural Guidelines](https://github.com/scg-unibe-ch/ese2016/wiki/Architectural-Guidelines) we need to: "All requests to controller objects should have logging support. A log entry should be added when a controller receives a request and when the controller completes a request. For example if a request is done to /profile/placeAd there should be a log entry that the controller PlaceAdController received this request and that the controller handled the request (successfully or with an error). Long entries should be placed in the file controller.log. Log entries should contain relevant information related to the request. Log entries should follow a common structure."

TL;DR: We need a Logger which:
-logs successful access to controllers or erros
-to a file called **controller.log**
-fills the a log-entry with relevant information related to the request itself (regardless of success/failure).

In order to do so we need:
1. Create a **Logger** which can be given a state-model, which itself contains information about the request itself.
2. A flag to indicate whether the request was successful or, if not, a error code
3. to create a common template so that the logger follows a common strucuter.

Implementation propotions:
1. Create a **Logger.java** constructor which can be given either a **Request**(which natively exists i.e. javax.servlet.HTTPRequest) or, if not, an Object, which contains important information about the request itself. Also the **Logger.java** will be given a state-object (proposed is an **enum: RequestState**, for various error codes or success),
2. The **Logger.java** contains a certain template which produces the log, which itself will be written into the file **controller.log**.
3. The **Logger.java** takes the information from the **Request object** and creates, following the template from 2., the log-message.
4. The **Logger.java** writes the log-message to the **controller.log**.

Please share your opinion on my proposed implementation and understanding of the task as I, myself, am not entirely sure whether I entirely understood the task and also as I am not a savant.

**Comment by remoroethlisberger** [comment]
This is NOT implemented, since it is not required as discussed with Claudio.
